src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static nature() {
+    this.alive = this.alive.filter((dead) =>
+      dead.health !== 0);
+  }
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100, toHide = false) {
+    super(name, health);
+    this.hidden = toHide;
+
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    Animal.alive.push(this);
+  }
+  bite(mammal) {
+    if (mammal.hidden === false) {
+      mammal.health -= 50;
+
+      if (mammal.health === 0) {
```

Comment:
The removal of dead animals from the `Animal.alive` array should not rely on the health being exactly `0`. It should be less than or equal to `0` to handle cases where health might drop below zero. Also, this check should be performed after every time an animal's health is modified, not just in the `bite` method.